packages/devextreme/js/__internal/ui/html_editor/ui/m_baseDialog.ts

```diff
@@ -0,0 +1,95 @@
+import devices from '@js/core/devices';
+import type { dxElementWrapper } from '@js/core/renderer';
+import $ from '@js/core/renderer';
+import type { DeferredObj } from '@js/core/utils/deferred';
+import { Deferred } from '@js/core/utils/deferred';
+import {
+  // @ts-expect-error
+  getCurrentScreenFactor,
+  hasWindow,
+} from '@js/core/utils/window';
+import type HtmlEditor from '@js/ui/html_editor';
+import type { Properties as PopupProperties } from '@js/ui/popup';
+import Popup from '@js/ui/popup';
+
+const DROPDOWN_EDITOR_OVERLAY_CLASS = 'dx-dropdowneditor-overlay';
+
+const isSmallScreen = (): boolean => {
+  const screenFactor = hasWindow() ? getCurrentScreenFactor() : null;
+  return devices.real().deviceType === 'phone' || screenFactor === 'xs';
+};
+abstract class BaseDialog<T = unknown> {
+  _editorInstance: HtmlEditor;
+
+  _popupUserConfig?: PopupProperties;
+
+  _popup!: Popup;
+
+  deferred?: DeferredObj<T>;
+
+  constructor(editorInstance: HtmlEditor, popupConfig?: PopupProperties) {
+    this._editorInstance = editorInstance;
+    this._popupUserConfig = popupConfig;
+
+    this._renderPopup();
+  }
+
+  protected _renderPopup() {
+    const editorInstance = this._editorInstance;
+    const $container = $('<div>')
+      .addClass(this._getPopupClass())
+      .appendTo(editorInstance.$element());
+    const popupConfig = this._getPopupConfig();
+
+    // @ts-expect-error
+    return editorInstance._createComponent($container, Popup, popupConfig);
```

Comment:
You only use EditorInstance to create a Popup. I am not sure if this is the correct solution. Maybe you should consider calling the new Popup constructor, as is done here: https://github.com/DevExpress/DevExtreme/blob/25_1/packages/devextreme/js/__internal/ui/m_dialog.ts#L87.

As a last resort you can do a type conversion, but I don't recommend it: editorInstance as any